### PR TITLE
Add Alliance to Who Command

### PR DESCRIPTION
**Pull request by UnlimitedFrames**

### CODE REVIEW

File: bot/src/embeds/pnw.rs [hunk 1]
```diff
@@ -21,15 +21,12 @@ pub fn alliance<'a>(
...
-    let cities = ctx.data().cache.find_many_cities(|a| {
-        nations.contains(
-            &ctx.data()
-                .cache
-                .find_exactly_one_nation(|n| n.id == a.nation_id)
-                .unwrap(),
-        )
-    });
 
+    let nation_ids = nations.iter().map(|a| a.id).collect::<Vec<i32>>();
```
Comment (mrvillage) [inline bot/src/embeds/pnw.rs:25]: Maybe use a HashSet here, O(1) contains instead of O(n)

File: bot/src/commands/who.rs [hunk 2]
```diff
@@ -7,31 +8,18 @@ use crate::types::{Context, Error};
 async fn who(
     ctx: Context<'_>,
     #[description = "Search for a nation or alliance"] search: String,
+    #[description = "If to search for an alliance"] is_alliance: Option<bool>,
```
Comment (mrvillage) [inline bot/src/commands/who.rs:11]: This works for now, in the future I'd like to create a union enum to facilitate a cleaner user experience.

**Review by mrvillage — CHANGES_REQUESTED** [review]
Looks great! Had one small bit of feedback and one note for the future. :)

File: bot/src/commands/who.rs (hunk 2 above)
Comment (UnlimitedFrames) [reply bot/src/commands/who.rs:11]: for sure, this command isnt finished, i just wanted to move on to somthing else :P

File: bot/src/embeds/pnw.rs (hunk 1 above)
Comment (UnlimitedFrames) [reply bot/src/embeds/pnw.rs]: done

**Review by mrvillage — APPROVED** [review]

**Pull request merged**